token-group/interface/src/instruction.rs

```diff
@@ -0,0 +1,337 @@
+    },
+};
+
+/// Instruction data for initializing a new `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:initialize_group")]
+pub struct InitializeGroup {
+    /// Update authority for the group
+    pub update_authority: OptionalNonZeroPubkey,
+    /// The maximum number of group members
+    pub max_size: u32,
+}
+
+/// Instruction data for updating the max size of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:update_group_max_size")]
+pub struct UpdateGroupMaxSize {
+    /// New max size for the group
+    pub max_size: u32,
+}
+
+/// Instruction data for updating the authority of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:update_group_authority")]
+pub struct UpdateGroupAuthority {
+    /// New authority for the group, or unset if `None`
+    pub new_authority: OptionalNonZeroPubkey,
+}
+
+/// Instruction data for initializing a new `Member` of a `Group`
+#[repr(C)]
+#[derive(Clone, Copy, Debug, PartialEq, Pod, Zeroable, SplDiscriminate)]
+#[discriminator_hash_input("spl_token_group_interface:initialize_member")]
+pub struct InitializeMember {
+    /// The pubkey of the `Group`
+    pub group: Pubkey,
+    /// The member number
+    pub member_number: u32,
+}
+
+/// All instructions that must be implemented in the SPL Token Group Interface
+///
+/// Note: Any instruction can be extended using additional required accounts by
+/// using the `InitializeExtraAccountMetaList` instruction to write
+/// configurations for extra required accounts into validation data
+/// corresponding to an instruction's unique discriminator.
+#[derive(Clone, Debug, PartialEq)]
+pub enum TokenGroupInterfaceInstruction {
+    /// Initialize a new `Group`
+    ///
+    /// Assumes one has already initialized a mint for the
+    /// group.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[]`   Mint
+    ///   2. `[s]`  Mint authority
+    InitializeGroup(InitializeGroup),
+
+    /// Update the max size of a `Group`
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[s]`  Update authority
+    UpdateGroupMaxSize(UpdateGroupMaxSize),
+
+    /// Update the authority of a `Group`
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Group
+    ///   1. `[s]`  Current update authority
+    UpdateGroupAuthority(UpdateGroupAuthority),
+
+    /// Initialize a new `Member` of a `Group`
+    ///
+    /// Assumes the `Group` has already been initialized,
+    /// as well as the mint for the member.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    ///   0. `[w]`  Member
```

Comment:
Interesting, if I'm reading this right, since this can just be any account, then a mint can be a member of multiple groups. Is that correct?

Reply:
Yep, if that's what we want! Right now I left it up for discussion.

Reply:
awesome, that works!